### PR TITLE
Preload schools for auto-complete

### PR DESCRIPTION
**Pull request by jopolsky**
Added an initializer for the schools auto-complete.

I wanted to log how long it took to preload. I couldn't seem to get `Rails.logger` working so I just used `puts`.

Also, I had to add a flag to each environment config, is there a master definition for config values?

### CODE REVIEW

**Comment by joannepolsky** [comment]
Humm, do initializers run before any DB creation/seeding? I'm assuming so, circle is blowing up on:

```
[2017-09-20 03:46:15] mysql://root@localhost/dashboard_test
[2017-09-20 03:46:15] RAILS_ENV=test RACK_ENV=test bundle exec rake db:create db:test:prepare
rake aborted!
ActiveRecord::StatementInvalid: Mysql2::Error: Table 'dashboard_test.schools' doesn't exist: SELECT `schools`.* FROM `schools`
...
/home/circleci/project/lib/cdo/school_autocomplete.rb:9:in `initialize'
/home/circleci/project/dashboard/config/initializers/preload_schools.rb:9:in `<top (required)>'
```

File: dashboard/config/initializers/preload_schools.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'cdo/school_autocomplete'
+
+# Preload the SchoolAutocomplete singleton instance.
+# Skip if this is running a Rake task (e.g. rake db:setup) or when caching is disabled
+
+if File.basename($0) != 'rake' && !Rails.application.config.skip_preload_schools
+  startTime = Time.now
+  SchoolAutocomplete.instance
+  endTime = Time.now
+  puts "[#{endTime.strftime('%c')}] Preloaded schools in #{endTime - startTime} second(s)"
```
Comment (jeremydstone) [inline dashboard/config/initializers/preload_schools.rb]: Pls use `CDO.log.info` rather than `puts`

**Comment by jeremydstone** [comment]
Generally looks good modulo removing the `puts`. Before merging I'd like @wjordan to weigh in on operational concerns. Will, preloading the 100K schools takes about 15 seconds on Joanne's machine. Do we have any concerns about adding 15 seconds to our current application launch process in production? (Presently, we believe it takes at least a minute, I think?) 

I don't have concerns as long as the instance isn't passing its health check and serving traffic until after the school preload happens. Since this is a Rails initializer, I believe this will have the behavior we want - it will take an additional 15 secs before the instance is "up" and active the load balancer, but won't be doing this work while traffic is going to it.

File: dashboard/config/initializers/preload_schools.rb (hunk 1 above)
Comment (jopolsky) [reply dashboard/config/initializers/preload_schools.rb]: Updated. I wasn't clear before that the logs were going to `dashboard/log`. Thanks!

File: dashboard/config/environments/development.rb [hunk 2]
```diff
@@ -59,6 +59,7 @@
   # Whether or not to skip script preloading. Setting this to true
   # significantly speeds up server startup time
   config.skip_script_preload = true
+  config.skip_preload_schools = false
```
Comment (wjordan) [inline dashboard/config/environments/development.rb]: shouldn't this default to `true` in `development` env? Most people running development server won't be working on the auto-complete feature, so the extra ~15 seconds on every start-up would not be ideal in this environment.

Comment (jopolsky) [reply dashboard/config/environments/development.rb]: Sure. I will update

**Comment by jopolsky** [comment]
@wjordan I've updated the PR now.

**Pull request merged**